collect/cache/cache.go

```diff
@@ -191,3 +193,84 @@ func (d *DefaultInMemCache) RemoveTraces(toDelete generics.Set[string]) {
 		}
 	}
 }
+
+// TraceCache uses an LRU cache to store traces
+// It does not control the maximum size of the cache, instead allowing it to grow indefinitely.
+// The collector is responsible for managing the cache size and removing traces when necessary.
+type TraceCache struct {
+	Metrics metrics.Metrics
+	Logger  logger.Logger
+
+	cache *lru.Cache[string, *types.Trace]
+}
+
+var _ Cache = (*TraceCache)(nil)
+
+var traceCacheMetadata = []metrics.Metadata{
+	{Name: "collect_cache_entries", Type: metrics.Histogram, Unit: metrics.Dimensionless, Description: "The number of traces currently stored in the cache"},
+}
+
+func NewTraceCache(metrics metrics.Metrics, logger logger.Logger) *TraceCache {
+	logger.Debug().Logf("Starting TraceCache")
+	defer func() { logger.Debug().Logf("Finished starting TraceCache") }()
+
+	for _, metadata := range traceCacheMetadata {
+		metrics.Register(metadata)
+	}
+
+	// allow the cache to grow really large by using math.MaxInt32 (2147483647) as cache size
+	cache, err := lru.New[string, *types.Trace](math.MaxInt32)
+	if err != nil {
+		logger.Error().Logf("Failed to create LRU cache: %s", err)
+		return nil
+	}
+
+	return &TraceCache{
+		Metrics: metrics,
+		Logger:  logger,
+		cache:   cache,
+	}
+}
+
+func (c *TraceCache) GetCacheCapacity() int {
+	return math.MaxInt32
```

Comment:
Yes, the cache capacity value was used to determine how full the cache was when using a ring buffer - ie how many free slots were available. The cache capacity was then used to apply other actions.

This cache implementation doesn't have a maximum capacity, instead it will allow the cache to grow and relies on the collect's memory usage check to shed load if neccesary.